x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/matrix_histogram/dns/query.dns_histogram.dsl.ts

```diff
@@ -89,7 +87,7 @@ export const buildDnsHistogramQuery = ({
         dns_name_query_count: {
           terms: {
             field: stackByField,
-            size: HUGE_QUERY_SIZE,
+            size: 10,
```

Comment:
As for the HUGE_QUERY_SIZE here, we used that to make sure we get the correct buckets order.
We first get the buckets for the top ten domains ordered by the doc_count, but eventually we sort them by the cardinality of `dns.question.name`. I was worried that if I didn't get enough size of terms bucket, some domain with smaller doc_count but with larger cardinality of dns.question.name might be dropped.
Could try reducing the HUGE_QUERY_SIZE to smaller count, like 100 and see if that works?

Reply:
Thanks for the context and side help. I think this query should be better now and much clearer without missing buckets from previous tickets. Let me know!